naps/cores/peripherals/soc_memory.py

```diff
@@ -66,7 +66,8 @@ def elaborate(self, platform):
             return m
 
         memorymap = MemoryMap()
-        memorymap.allocate("memory", writable=True, bits=self.depth * memorymap.bus_word_width)
+        assert memorymap.bus_word_width == 32
```

Comment:
`>= 32` should be fine here afaik

Reply:
I think it has to be exactly 32, assuming `memorymap.bus_word_width` is exactly the width of `data` in `handle_read` and `handle_write`.

Reply:
why?

Reply:
because the 32 in those functions would have to be changed to `memorymap.bus_word_width`